mantis-runtime/src/main/java/io/mantisrx/runtime/source/http/impl/HttpSourceImpl.java

```diff
@@ -708,7 +693,11 @@ public void serverRemoved(ServerInfo server) {
         }
 
         public void reset() {
-            connectedServers.clear();
+            Set<ServerInfo> connectedServerInfos = connectedServers.keySet();
```

Comment:
LGTM, only concern is that iterating over keyset may throw exception if map is changed. i'm not familiar enough with concurrent hashmap to know if that can happen or not. you can also just use getConnectedServers()

Reply:
@calvin681 I can look into the behavior of keySet() under changes to the underlying concurrent map but do you think removeConnectedServer where we close the underlying HttpClient makes sense?

Reply:
According to the keyset() doc the iterator is 'weakly consistent' and shall never emit modification exception.
In the meantime, I wonder if there is anything blocking/timeout inside the httpclient shutdown call that in extreme cases can cause this operation to hang/stuck for a long time?

Reply:
If so maybe this can be done as fire-and-forget with timeout in a different thread?